app/Filters/CreditFilters.php

```diff
@@ -0,0 +1,193 @@
+<?php
+
+/**
+ * Invoice Ninja (https://invoiceninja.com)
+ *
+ * @link https://github.com/invoiceninja/invoiceninja source repository
+ *
+ * @copyright Copyright (c) 2020. Invoice Ninja LLC (https://invoiceninja.com)
+ *
+ * @license https://opensource.org/licenses/AAL
+ */
+
+namespace App\Filters;
+
+use App\Models\Credit;
+use App\Models\Invoice;
+use App\Models\User;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Support\Carbon;
+use Illuminate\Support\Facades\Log;
+
+class CreditFilters extends QueryFilters
+{
+    /**
+     * Filter based on client status
+     *
+     * Statuses we need to handle
+     * //todo ?partials as a status?
+     * - all
+     * - paid
+     * - unpaid
+     * - overdue
+     * - reversed
+     *
+     * @param      string client_status The credit status as seen by the client
+     * @return     Illuminate\Database\Query\Builder
+     *
+     */
+    
+    public function client_status(string $value = '') :Builder
+    {
+        if (strlen($value) == 0) {
+            return $this->builder;
+        }
+
+        $status_parameters = explode(",", $value);
+
+        if (in_array('all', $status_parameters)) {
+            return $this->builder;
+        }
+        
+        if (in_array('paid', $status_parameters)) {
+            $this->builder->where('status_id', Credit::STATUS_PAID);
+        }
+
+        if (in_array('unpaid', $status_parameters)) {
```

Comment:
Need to change these to suit credits...

Credit::STATUS_PARTIAL 


in_array('partial'